docs/openapi.yaml

```diff
@@ -0,0 +1,1143 @@
+      description: |
+        ### Overview
+        Lookup map of the area where the robot is located
+      tags:
+        - Robots
+      parameters:
+        - in: path
+          name: robot_id
+          required: true
+          schema:
+            type: integer
+            format: int32
+            example: 21
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Map"
+          description: Request successful
+        "404":
+          description: Not found - No robot with id exists
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+
+  /robots/{robot_id}/schedule:
+    get:
+      operationId: getRobotSchedule
+      summary: Lookup the schedule for the robot
+      description: |
+        ### Overview
+        Get the current schedule of a robot. The schedule is a list of time entries where the robot is scheduled to perform a certain mission.
+
+        The minimum start time and maximum end time for the schedule entries can be specified in the query. If none is provided, the default start time is current time and default end time is start time + one week.
+      tags:
+        - Robots
+      parameters:
+        - in: path
+          name: robot_id
+          required: true
+          schema:
+            type: integer
+            format: int32
+            example: 21
+        - in: query
+          name: startTime
+          schema:
+            type: string
+            format: date-time
+            description: Minimum start time for scheduled missions
+          required: false
+        - in: query
+          name: endTime
+          schema:
+            type: string
+            format: date-time
+            description: Maximum end time for scheduled missions
+          required: false
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Schedule"
+          description: Request successful
+        "404":
+          description: Not found - No robot with id exists
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetails"
+
+  /robots/{robot_id}/event:
+    post:
+      operationId: postRobotEvent
+      summary: Create new event for robot
+      description: |
+        ### Overview
+        Adds a new event to the robot schedule
```

Comment:
Is task a better word than event?

Reply:
No. We already have a concept called Task

Reply:
Would a task be one part of the event (one step of the mission)? If so could mission be a better word? I struggle to see what an event would be?

Reply:
Mission is used about the Echo Missions. Event is used about starting a particular echo mission at a given time.